1InterviewOddEvenSeparation.java

```diff
@@ -0,0 +1,49 @@
+//Given an array odd numbers should come left side and even number should come right side. (Do inplace shifting meaning use the same input array dont create any new array for storing the output)
+//        Input: [1,2,3,4,5,6]
+//        Output: [1,3,5,2,4,6]
+
+import java.util.Scanner;
+class Pointers{
+    public void separateOddEven(int[] a){
+        int left=0,right=a.length-1; // left and right pointers
+        while(left<right){
+            if (a[left] % 2 != 0 && a[right] % 2 != 0) left++;
+            else if (a[left] % 2 != 0 && a[right] % 2 == 0) {
+                left++;
+                right--;
+            } else if (a[left] % 2 == 0 && a[right] % 2 != 0) {
+                int temp=a[right];
+                a[right]=a[left];
+                a[left]=temp;
+                left++;
+                right--;
+            } else {
+                if (left - right == 1) {
+                    left++;
+                    right--;
+                }
+                else right--;
+            }
+        }
```

Comment:
In case of counting the swaps we can add a count variable and find the minimum number of swapes